quinn-proto/src/timer.rs

```diff
@@ -23,58 +19,51 @@ pub(crate) enum TimerKind {
     KeepAlive = 5,
 }
 
-impl TimerKind {
-    const VALUES: [Self; 6] = [
-        TimerKind::LossDetection,
-        TimerKind::Idle,
-        TimerKind::Close,
-        TimerKind::KeyDiscard,
-        TimerKind::PathValidation,
-        TimerKind::KeepAlive,
+impl Timer {
+    pub(crate) const VALUES: [Self; 6] = [
+        Timer::LossDetection,
+        Timer::Idle,
+        Timer::Close,
+        Timer::KeyDiscard,
+        Timer::PathValidation,
+        Timer::KeepAlive,
     ];
 }
 
 /// A table of data associated with each distinct kind of `Timer`
 #[derive(Debug, Copy, Clone, Default)]
-pub struct TimerTable<T> {
+pub(crate) struct TimerTable<T> {
```

Comment:
Is the generic `T` here still pulling its weight?

Reply:
Definitely not.